File: frontend/src/js/services/diagnosis.service.js
/**
 * Created by tatiana.gorbunova on 09.04.2017.
 */
module.exports = (app) => {

    require('./globalvar.service')(app);

    app.factory('DiagnosisService', ['$resource', 'GlobalVarService', function($resource, GlobalVarService) {
        return $resource(GlobalVarService.BACKEND_URL + '/diagnosis', {}, {
            list: {
                method: 'GET',
                isArray: true
            }
        });
    }])
}

File: backend/src/main/java/ru/tatiana/demo/service/DiagnosisService.java
package ru.tatiana.demo.service;

import ru.tatiana.demo.model.Diagnosis;

import java.util.List;

/**
 * Created by tatiana.gorbunova on 24.12.2016.
 */
public interface DiagnosisService {

    List<Diagnosis> getDiagnoses(Integer offset, Integer pageSize);

    Diagnosis getById(Long id);
}
